pkg/network/java/testutil/JavaClientSimulator/src/main/java/Clients/JavaClients.java

```diff
@@ -19,118 +21,149 @@
+                System.out.println("Executing handler for Apache Http client:");
+                callback = () -> {
+                    try {
+                        JavaClients.HTTPApacheClientExample(url);
+                    } catch (IOException e) {
+
+                    }
+                };
+                executeCallbackLogic(iterations, sleepInterval, url, callback);
+                break;
+            case okhttp:
+                System.out.println("Executing handler for OkHttp client:");
+                callback = () -> {
+                    try {
+                        JavaClients.OkHttpClient(url);
+                    } catch (IOException e) {
+
+                    }
+                };
+                executeCallbackLogic(iterations, sleepInterval, url, callback);
+                break;
+            case httpclient:
+                System.out.println("Executing handler for HttpClient client:");
+                callback = () -> {
+                    try {
+                        JavaClients.HTTPClient(url);
+                    } catch (IOException e) {
+
+                    }
+                };
+                executeCallbackLogic(iterations, sleepInterval, url, callback);
+                break;
+            case urlconnection:
+                System.out.println("Executing handler for URLConnection client:");
+                callback = () -> {
+                    try {
+                        JavaClients.HttpsURLConnection(url);
+                    } catch (IOException e) {
+
+                    }
+                };
+                executeCallbackLogic(iterations, sleepInterval, url, callback);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported callback type: " + clientType);
         }
     }
 
-    public static void HTTPClientExample(int iterations, String url) throws InterruptedException {
-        if (iterations == -1){
-            while (true){
-                try {
-                    HttpClient httpClient = HttpClient.newHttpClient();
-                    HttpRequest request = HttpRequest.newBuilder()
-                            .uri(URI.create(URL_SCHEME+url))
-                            .version(HttpClient.Version.HTTP_1_1)
-                            .build();
-                    HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
-                    System.out.println("Response " + response.toString());
-                } catch (IOException | InterruptedException e) {
-                    e.printStackTrace();
-                }
-                finally {
-                    Thread.sleep(5000);
+    private static void executeCallbackLogic(int iterations, long sleepInterval, String url, Runnable callback) throws InterruptedException {
+        if (iterations == -1) {
+            // Infinite loop
+            while (true) {
+                callback.run();
```

Comment:
```suggestion
                    try {
                        callback.run();
                    } catch (IOException e) {

                    }
```

Reply:
that is invalid, `run` doesn't throw IOException